docs/user-guide/building-assistants.rst

```diff
@@ -0,0 +1,1018 @@
+   * faq: What’s Rasa X?
+     - respond_faq
+
+You can read more in this `blog post <https://blog.rasa.com/response-retrieval-models/>`_ and the
+`Retrieval Actions <https://rasa.com/docs/rasa/core/retrieval-actions/>`_ page.
+
+Using the features we described in this tutorial, you can easily build a context-less assistant.
+When you’re ready to enhance your assistant with context, check out :ref:`build-contextual-assistant`.
+
+.. _build-contextual-assistant:
+
+Building contextual assistants
+------------------------------
+
+Whether you’ve just created an FAQ bot or are starting from scratch, the next step is to expand
+your bot to handle contextual conversations. 
+
+In this tutorial we’re going to cover a variety of topics:
+
+    - Handling business logic
+    - Handling unexpected user input
+    - Failing gracefully
+    - More complex contextual conversations
+
+Please make sure you’ve got all the data from the :ref:`build-faq-assistant` before starting this tutorial.
+You will need to make some adjustments to your configuration file, since we now need to pay attention to context: 
+
+.. code-block:: yaml
+
+   policies:
+   - name: MemoizationPolicy
+   - name: MappingPolicy
+
+We removed the ``max_history: 1`` configuration, because the default is ``5``, meaning Core will pay
+attention to the past 5 turns when making a prediction (see explanation of `max history <https://rasa.com/docs/rasa/core/policies/#max-history>`_).
+
+Business logic
+^^^^^^^^^^^^^^
+
+A lot of conversational assistants have user goals that involve collecting a bunch of information
+from the user before being able to do something for them. This is called slot filling. For
+example, in the banking industry you may have a user goal of transferring money, where you
+need to collect information about which account to transfer from, whom to transfer to and the
+amount to transfer. This type of behaviour can and should be handled in a rule based way, as it
+is clear how this information should be collected. 
+
+For this type of use case, we can use Forms and our FormPolicy. The `FormPolicy <https://rasa.com/docs/rasa/core/policies/#form-policy>`_
+works by predicting the form as the next action until all information is gathered from the user. 
+
+As an example, we will build out the SalesForm from Sara. The user wants to contact
+our sales team, and for this we need to gather the following pieces of information:
+
+    - Their job 
+    - Their bot use case
+    - Their name
+    - Their email 
+    - Their budget
+    - Their company
+
+We will start by defining the ``SalesForm`` as a new class in the file called ``actions.py``.
+The first method we need to define is the name, which like in a regular Action
+returns the name that will be used in our stories:
+
+.. code-block:: python
+
+   from rasa_sdk.forms import FormAction
+
+   class SalesForm(FormAction):
+       """Collects sales information and adds it to the spreadsheet"""
+
+       def name(self):
```

Comment:
Optional: we could add type annotations to `name` and `required_slots` (since they are present for `submit`)